python_modules/dagster/dagster/_core/remote_representation/code_location.py

```diff
@@ -111,25 +115,82 @@ def get_repository_names(self) -> Sequence[str]:
     def name(self) -> str:
         return self.origin.location_name
 
-    @abstractmethod
     def get_execution_plan(
         self,
         remote_job: RemoteJob,
         run_config: Mapping[str, object],
         step_keys_to_execute: Optional[Sequence[str]],
         known_state: Optional[KnownExecutionState],
         instance: Optional[DagsterInstance] = None,
-    ) -> RemoteExecutionPlan: ...
+    ):
+        execution_plan_snapshot_without_job_snapshot_id = (
+            self.get_execution_plan_snapshot_without_job_snapshot_id(
+                remote_job.get_remote_origin(),
+                remote_job.get_subset_selector(
+                    remote_job.asset_selection,
+                    remote_job.asset_check_selection,
+                ),
+                run_config,
+                step_keys_to_execute,
+                known_state,
+                instance,
+            )
+        )
+
+        return RemoteExecutionPlan(
+            execution_plan_snapshot=execution_plan_snapshot_without_job_snapshot_id._replace(
+                job_snapshot_id=remote_job.identifying_job_snapshot_id,
+            )
+        )
 
     @abstractmethod
+    def get_execution_plan_snapshot_without_job_snapshot_id(
```

Comment:
im not sure where is best but we should put doc blocks in the two methods to describe the motivation behind the difference